### PR TITLE
Add symbol scope to legend item

### LINKED ISSUE #31756: [layouts] @symbol_label variable is not listed in the expression dialog, no help
Add a legend item to a map layout
In the 'Legend items' section, select a layer and click the "Add an expression to the vector layer and each child symbol's label"
The expression dialog opens with a symbol `@symbol_label` already entered
What that variable represents is not available anywhere. I think it should be listed next to the other variables.

master 3.9 cdd3a55e9e

### PR DESCRIPTION
**Pull request by m-kuhn**
Fix #31756

### CODE REVIEW

File: src/app/layout/qgslayoutlegendwidget.cpp [hunk 1]
```diff
@@ -1030,6 +1031,29 @@ void QgsLayoutLegendWidget::mLayerExpressionButton_clicked()
...
+
+  if ( r )
+  {
+    const QgsLegendSymbolList legendSymbols = r->legendSymbolItems();
+
+    if ( !legendSymbols.empty() )
+    {
+      QgsSymbolLegendNode legendNode( layerNode, legendSymbols.first() );
+
+      symbolLegendScope->addVariable( QgsExpressionContextScope::StaticVariable( QStringLiteral( "symbol_label" ), symbolLabel().remove( "[%" ).remove( "%]" ), true ) );
```
Comment (m-kuhn) [inline src/app/layout/qgslayoutlegendwidget.cpp]: @roya0045 could you come up with the code required to reimplement the `symbolLayer()` logic in here?

Comment (roya0045) [reply src/app/layout/qgslayoutlegendwidget.cpp]: You mean `symbolLabel()` or `symbolLayer()`?

Comment (roya0045) [reply src/app/layout/qgslayoutlegendwidget.cpp]: simply do `legendNode.symbolLabel().remove( "[%" ).remove( "%]" )`
The function is a public member of the `QgsSymbolLegendNode` see [this line](https://github.com/roya0045/QGIS_1/blob/19d37b4dc95fee889d83d3d114c1d6cdc621560b/src/core/layertree/qgslayertreemodellegendnode.h#L336)

Comment (m-kuhn) [reply src/app/layout/qgslayoutlegendwidget.cpp]: Thanks.

**Comment by roya0045** [comment]
Thanks Matthias! And sorry that you had to this, it was an oversight on my end.

**Pull request merged**